assignment/scripts/function-practice.js

```diff
@@ -50,40 +60,123 @@ console.log( 'isPositive - should say false', isPositive(-3) );
+    if (x === value) {
+      return true;
+    }
+  }
+  return false;
 }
 
+console.log(`This is a test of 3, [2, 3, 4] and should return true: ${find(3, [2, 3, 4])}`);
+console.log(`This is a test of 7, [4, 5, 6] and should return false: ${find(7, [4, 5, 6])}`);
+
+
+
 // ----------------------
 // Stretch Goals
 // ----------------------
 // 8. Function to check if a letter is the first letter in a 
 //    string. Return true if it is, and false otherwise
 function isFirstLetter(letter, string) {
+  startLetter = string[0];
+  if (letter === startLetter) {
+    return true;
+  }
+    return false;
 
 }
 console.log( 'isFirstLetter - should say true', isFirstLetter('a', 'apple') );
 console.log( 'isFirstLetter - should say false', isFirstLetter('z', 'apple') );
+console.log( 'isFirstLetter - should say true', isFirstLetter('t', 'the brown dog') );
+
 
 // 9. Function to return the sum of all numbers in an array
-function sumAll( ) {
-  let sum = 0
+function sumAll(numArray) {
+  let sum = 0;
   // TODO: loop to add items
+  for (x of numArray) {
+    sum += x;
+  }
   return sum;
 }
 
+console.log(`Test - sumAll should return 15: ${sumAll([3, 9, 3])}`);
+console.log(`Test - sumAll should return 18.5: ${sumAll([14, 2.5, 2])}`);
+
+
 // 10. Function to return a new array of all positive (greater than zero)
 //     numbers contained in an input array. If there are no positive numbers
 //     return an empty array. Note: The input array should not change.
 
+function posOnly (array) {
+
+  let posArray = [];
+
+  for (x of array) {
+    if (x > 0) {
+      posArray.push(x);
+    }
+  }
+
+  return posArray;
+
+}
+
+console.log(`posOnly with array [2, 3, 4] should return [2, 3, 4]: ${posOnly([2, 3, 4])}`);
+console.log(`posOnly with array [2, 3, -4] should return [2, 3]: ${posOnly([2, 3, -4])}`);
+console.log(`posOnly with array [-2, -3, -4] should return []: ${posOnly([-2, -3, -4])}`);
+console.log(`posOnly with array [0, -1, 4] should return [4]: ${posOnly([0, -1, 4])}`);
 
 
 // 11. Pick a problem from Edabit(https://edabit.com/) or 
 //     CodeWars(https://www.codewars.com/). Then describe it 
 //     here in a comment, write the function, and test it!
+
+/* Codewars - Disemvowel Trolls
+** The purpose of this function is to combat trolls by
+** removing each vowel from their comment. This function
+** will accept a string as an input, and return a string 
+** with the vowels removed. For the purposes of this
+** problem, y is not a vowel.
+*/
+
+/* I originally did this using a for in loop to iterate over the string,
+** then a for of loop to iterate over each vowel. If the value of str at
+** index x was equal to any value in vowels, I
+** then used str.replace. This worked, but at a certain string
+** length it would remove an index != the vowel/string index I wanted to remove.
+** I had trouble figuring this out, so looked up ways to see if a value is in an array,
+** and found the array.includes method...and that was a lot easier.
```

Comment:
Includes is a nice solution for this. You could do something like `x === 'a' || x === 'e' || x === ...`, That is super long, but it would be _manageable_, particularly if you put it in a separate `isVowel` function.